omero_figure/scripts/omero/figure_scripts/Figure_To_Pdf.py

```diff
@@ -1677,6 +1676,234 @@ def draw_scalebar(self, panel, region_width, page):
                 font_size, (red, green, blue),
                 align="center")
 
+    def get_color_ramp(self, channel):
+        """
+        Return the 256 1D array of the LUT from the server or
+        the color gradient.
+
+        LUT files on the server are read with the script service, and
+        file content is parsed with a custom implementation.
+        """
+        color = channel["color"]
+
+        # Convert the hexadecimal string to RGB
+        color_ramp = None
+        if len(color) == 6:
+            try:
+                r = int(color[0:2], 16)
+                g = int(color[2:4], 16)
+                b = int(color[4:6], 16)
+                color_ramp = (numpy.linspace(0, 1, 256).reshape(-1, 1)
+                              * numpy.array([r, g, b], dtype=numpy.uint8).T)
+                color_ramp = color_ramp.astype(numpy.uint8)
+            except ValueError:
+                pass
+
+        else:
+            script_service = self.conn.getScriptService()
+            luts = script_service.getScriptsByMimetype("text/x-lut")
+            self.conn.SERVICE_OPTS.setOmeroGroup(  # required to get LUT files
+                self.conn.getEventContext().groupId)
```

Comment:
Testing just now, I found that after creating a LUT colorbar, other images failed to load and were missing from the PDF.
So I realised that we need to `setOmeroGroup("-1")` after loading the originalFile, otherwise we might fail to load subsequent images.

Reply:
I could not reproduce, but fixed. Tell me if that worked

Reply:
Yes, that worked. You'll only see the bug if your images aren't in your default group.